SessionUIKit/Utilities/Localization+Style.swift

```diff
@@ -0,0 +1,235 @@
+        if stringLength > finalMatch.range.upperBound {
+            partsAndTags.append((attrString[finalMatch.range.upperBound...], Array(openTags.keys)))
+        }
+
+        /// Lastly we should construct the attributed string, applying the desired formatting
+        self.init(
+            attributedString: partsAndTags.reduce(into: NSMutableAttributedString()) { result, next in
+                result.append(NSAttributedString(string: next.part, attributes: next.tags.format(with: font)))
+            }
+        )
+    }
+
+    private subscript(range: Range<Int>) -> String {
+        attributedSubstring(from: NSMakeRange(range.lowerBound, (range.upperBound - range.lowerBound))).string
+    }
+
+    private subscript(range: PartialRangeFrom<Int>) -> String {
+        let upperBound: Int = self.string.utf16.count
+        return attributedSubstring(from: NSMakeRange(range.lowerBound, (upperBound - range.lowerBound))).string
+    }
+
+    private subscript(range: PartialRangeThrough<Int>) -> String {
+        attributedSubstring(from: NSMakeRange(0, range.upperBound)).string
+    }
+}
+
+private extension Collection where Element == NSAttributedString.HTMLTag {
+    func format(with font: UIFont) -> [NSAttributedString.Key: Any] {
+        func fontWith(_ font: UIFont, traits: UIFontDescriptor.SymbolicTraits) -> UIFont {
+            /// **Note:** Constructing a `UIFont` with a `size`of `0` will preserve the textSize
+            return UIFont(
+                descriptor: (font.fontDescriptor.withSymbolicTraits(traits) ?? font.fontDescriptor),
+                size: 0
+            )
+        }
+
+        return self.reduce(into: [NSAttributedString.Key: Any]()) { result, tag in
+            switch tag {
+                case .bold where self.contains(.italic), .italic where self.contains(.bold):
+                    result[.font] = fontWith(font, traits: [.traitBold, .traitItalic])
+
+                case .bold: result[.font] = fontWith(font, traits: [.traitBold])
+                case .italic: result[.font] = fontWith(font, traits: [.traitItalic])
+                case .underline: result[.underlineStyle] = NSUnderlineStyle.single.rawValue
+                case .strikethrough: result[.strikethroughStyle] = NSUnderlineStyle.single.rawValue
+                case .primaryTheme: result[.foregroundColor] = ThemeManager.primaryColor.color
+            }
+        }
+    }
+}
+
+public extension LocalizationHelper {
+    func localizedFormatted(in view: FontAccessible) -> NSAttributedString {
+        return localizedFormatted(baseFont: (view.fontValue ?? .systemFont(ofSize: 14)))
+    }
+    
+    func localizedFormatted(baseFont: UIFont) -> NSAttributedString {
+        return NSAttributedString(stringWithHTMLTags: localized(), font: baseFont)
+    }
+    
+    func localizedDeformatted() -> String {
+        return NSAttributedString(stringWithHTMLTags: localized(), font: .systemFont(ofSize: 14)).string
+    }
+
+}
```

Comment:
I moved this part to `SessionUIKit` only because if we want to support <span> we need to know the `UIColour` value of the theme primary colour, so we can put this value to the `foregroundColour` attribute. If we can get the colour value in `SessionUtilitiesKit`, we don't need to seperate this part of code as an extension in `SessionUIKit`. Right now I can't think of an idea on how I can get the `UIColour` value without touching things in `SessionUIKit`

Reply:
Hmm... ok in that case we can probably leave it here for now then I can look to do a similar change to what I currently have in groups rebuild - ie. the `SessionUtilitiesKit.configure` would have some closure(s) which can be used to retrieve the relevant theme values, this code would be shifted back into `LocalizationHelper` and would call those closure in order to retrieve the values when needed

I'm actually not 100% sold on the approach I've setup at the moment (seems like a lot of hassle to reduce the build time a little... 🥲)